pika/adapters/select_connection.py

```diff
@@ -238,51 +234,38 @@ def poll(self, write_only=False):
             the adapter can write.
 
         """
-        # Build our values to pass into select
-        input_fileno, output_fileno, error_fileno = [], [], []
-
-        if self.events & READ:
-            input_fileno = [self.fileno]
-        if self.events & WRITE:
-            output_fileno = [self.fileno]
-        if self.events & ERROR:
-            error_fileno = [self.fileno]
 
         # Wait on select to let us know what's up
         try:
-            read, write, error = select.select(input_fileno,
-                                               output_fileno,
-                                               error_fileno,
-                                               SelectPoller.TIMEOUT)
+            read, write, error = select.select(
+                self.r_fds, self.w_fds, self.e_fds, self.get_next_timeout())
         except select.error as error:
-            return self._handler(self.fileno, ERROR, error)
+            if error.errno != errno.EINTR:
+                raise
 
         # Build our events bit mask
-        events = 0
-        if read:
-            events |= READ
-        if write:
-            events |= WRITE
-        if error:
-            events |= ERROR
 
-        if events:
-            self._handler(self.fileno, events, write_only=write_only)
+        for fileno in self.filenos:
+            events = 0
+            if fileno in read:
+                events |= READ
+            if fileno in write:
+                events |= WRITE
+            if fileno in error:
+                events |= ERROR
```

Comment:
Not sure if it's an issue overall, but thought I'd mention it: in Mac OS `select`, the 3rd fd set arg is for errors, but on linux and unix (and windows?) it's for "exceptfds", which are for out-of-band data in case of TCP/IP (but not errors)